### ISSUE TITLE
[generate:entity:content] --has-bundles - Duplicate hook_theme

### ISSUE THREAD

**Issue by edysmp** [hechoendrupal/drupal-console#1562]
# Problem/Motivation

Create entity content with bundle duplicate **hook_theme**.
# How to reproduce
-  Drupal version  **8.0.0**
- Console version **0.9.10**
#### Steps
- Create a module, **generate:module**
- Create the entity with bundle, **generate:entity:content**

This produce:

``` php
/**
 * Implements hook_theme().
 */
function my_module_theme() {
  $theme = [];

  return $theme;
}

/**
 * Implements hook_theme().
 */
function my_module_theme() {
  $theme = [];
  $theme['default_entity'] = [
    'render element' => 'elements',
    'file' => 'default_entity.page.inc',
    'template' => 'default_entity',
  ];
  $theme['default_entity_content_add_list'] = [
    'render element' => 'content',
    'variables' => ['content' => NULL],
    'file' => 'default_entity.page.inc',
  ];
  return $theme;
}
```

**Comment by darol100** [comment]
@edysm , I have create a pull request (https://github.com/hechoendrupal/DrupalConsole/pull/1466) to remove **generate:module** because is empty. In the **generate:entity:content** it seem to be using the hook_theme.


However, that PR is been block until this issue been solve https://github.com/hechoendrupal/DrupalConsole/issues/1423, which gives you the ability of generating hook_theme stand alone.

**Comment by jmolivas** [comment]
@darol100: PR https://github.com/hechoendrupal/DrupalConsole/pull/1466 was merged, should we consider this pending or can this be closed?

**Comment by darol100** [comment]
**tl;dr**: No until we fix #1423 

This problem is been solve "kinda" from the **generate:module**. However, there still a problem. Because current way to generate the "hook_theme" does not take in consider if there is another hook_theme to and added the current value hook. Instead what it does is is generate a new a new hook_theme.

This been said... if the user run again **generate:entity:content** is going to generate a new hook_theme and is going to run into the same issue.

Here is an example...

![screen shot 2015-12-31 at 4 16 03 pm](https://cloud.githubusercontent.com/assets/6808460/12069249/e7e0e400-afd9-11e5-85b0-61f742f9c71f.png)

What is the solution ? We need to try fixed #1423 and after that we need to use that as chain where anywhere use are generating a hook_theme.

**Comment by UH-ESRM** [comment]
Not sure whether this is explicit in this thread. But as corroboration, if I generate the module and then generate entity content, as described above AND if I use the default when generating the module to output a theme AND I use the non-default choice to allow bundles with the generated content entity, then when I enable to the module, no entity options show up in the /admin/structure selections. However, if I do the same as above but I select NO THEME for the module, then everything works fine and I can see the two options - one for creating new bundle types and one for listing the bundle types. This is really great, but I can only see the entity options if I don't select the generation of a theme for the associated module.

**Comment by nequeteme** [comment]
HI @edysmp, I tested and I can't replicate, the script generates only one function, i tested with drupal/console 1.9.0 and after the run the steps 

- Create a module, generate:module
- Create the entity with bundle, generate:entity:content

the file  generated was

`/**
 * Implements hook_theme().
 */
function test1_theme() {
  return [
    'test1' => [
      'render element' => 'children',
    ],
  ];
}`

**Comment by enzolutions** [comment]
Thanks @nequeteme for your test, closing this issue.
